svg/renderer.go

```diff
@@ -397,3 +353,81 @@ func (r *SVG) RenderImage(img image.Image, m canvas.Matrix) {
 	r.writeClasses(r.w)
 	fmt.Fprintf(r.w, `"/>`)
 }
+
+// return a WriterTo, a refMask and a mimetype
+func (r *SVG) prepareImage(img image.Image) (func(io.Writer) error, string, canvas.ImageMimetype) {
```

Comment:
What do you think?

Reply:
You are right that this function is used only once and badly named.

I find it very convenient to have a dedicated function for this, because it allows to use early returns which makes it very readable.

I renamed it to `encodableImage` to better show its purpose : encode the image (if needed) to make it embeddable in the svg.
The only side-effect I am not really happy (but have no better solution yet) is the `renderOpacityMask` which does not have much to do in this function (but I can't find a better place for now).

> This also obviates the need to create the func(io.Writer) error return type, which is nice but a little too smart for what we need

I wasn't able to find an elegant solution without this "trick":
- the Mimetype is needed at the beginning for the `<image` tag
- the content of the image must be somehow encoded in base64

Since they depend on each other, this way we get the mimetype and the "WriterTo" at the same time (it's actually not an exact [`io.WriterTo`](https://golang.org/pkg/io/#WriterTo) )

---

If you have an elegant solution to solve this, I would be happy to integrate it in this PR!

Reply:
> The prepareImage serves no real purpose other than breaking up a long function into two, this scatters relevant code that should be close to each other into different functions, and also requires us to work with callbacks. Nothing against callbacks, but the design could be much more simple, so why go advanced?

The only other solution that comes to my mind is to add an intermediary buffer.

Instead of returning a `func(io.Writer) error`, the function can return `[]byte, error`.
It adds one error check more (after the call to the `encodableImage`) and is probably more memory hungry (this is a wild guess, and probably not relevant at this point in time in the project).

So if you think that adding an intermediary buffer is better, I would change this function (and if anyone is unhappy, he can make its own svg renderer, since the renderers are now independent from the code ^^)